src/pal/pal_openbsd.h

```diff
@@ -11,10 +11,23 @@
 
 namespace snmalloc
 {
-  class PALOBSD : public PALBSD
+  /**
+   * OpenBSD platform abstraction layer.
+   *
+   * OpenBSD behaves exactly like a generic BSD platform but this class exists
+   * as a place to add OpenBSD-specific behaviour later, if required.
+   */
+  class PALOBSD : public PALBSD<PALOBSD>
   {
   public:
-    static constexpr uint64_t pal_features = LazyCommit;
+    /**
+     * The features exported by this PAL.
+     *
+     * Currently, these are identical to the generic BSD PAL.  This field is
+     * declared explicitly to remind anyone who modifies this class that they
+     * should add any required features.
+     */
+    static constexpr uint64_t pal_features = PALBSD<PALOBSD>;
```

Comment:
Little mistake here should be `PALPOSIX<OS>::pal_features` or `PALBSD<OS>::pal_features`

Reply:
Just saw that while refactoring the Apple PAL.  Should be fixed now.

Reply:
Just a little idea if you wish for APPLE, you can either choose calling raw vm_allocate/vm_deallocate or even keeping mmap wrapper but adding VM_MAKE_FLAGS for the file descriptor arguments ... practical to monitor anonymous pages.